### DISCUSSION TITLE
Is `expo-dev-client` an HTTP client and is it proxy-aware?

### DISCUSSION THREAD

**Discussion by thelazyoxymoron** [expo/expo#18025] [Archived: Questions]
I'm facing a weird problem at work. We're using `BrowserStack` to test applications and since we don't have a physical iOS device, opting testing preview builds on their real device clouds. 

They offer [local testing mode](https://www.browserstack.com/docs/app-live/local-testing/test-using-local-testing) to run development builds. Steps that I'm trying to reproduce: 

1. create a custom dev client with our codebase
2. install that client on their device
3. enable local testing mode
4. start the application by `expo start --dev-client`

However, at this step, the app just keeps hanging on the splash screen. After having a lot of back and forth with BrowserStack's support team, they're asking whether the client that I'm using is proxy-aware/HTTP client. Below is the exact response from them:

> Can you please confirm if expo-dev-client is an HTTP client? I am adding some details about the HTTP client:
An HttpClient can be used to send requests and retrieve their responses. An HttpClient is created through a builder. The builder can be used to configure per-client state, like the preferred protocol version ( HTTP/1.1 or HTTP/2 ), whether to follow redirects, a proxy, an authenticator, etc. Once built, an HttpClient is immutable and can be used to send multiple requests.

> For e.g. [System.Net.Http](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-6.0) is the HTTP client used by Dart.

Anyone has any pointers? Thanks!

**Comment by micahlt** [comment]
I have this exact same question - did you ever figure this out @thelazyoxymoron?

**Reply by thelazyoxymoron** [reply]
It's been a long time, I don't remember arriving at a solution.
